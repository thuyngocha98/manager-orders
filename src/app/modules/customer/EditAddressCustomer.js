import React from 'react';
import {View, Text, TouchableOpacity, Keyboard} from 'react-native';
import styles from '@modules/customer/styles/editAddressCustomer.styles';
import Header from '@components/Header';
import I18n from '@assets/localization/I18n';
import IconGoBackArrow from '@assets/iconSvg/IconGoBackArrow';
import {responsiveHeight, responsiveWidth} from '@utils/DimenUtils';
import Colors from '@const/Colors';
import {Hoshi} from 'react-native-textinput-effects';
import IconCheckMark from '@assets/iconSvg/IconCheckMark';
import {useSelector, useDispatch} from 'react-redux';
import {reduxEditCustomer} from '@actions/customerAction';

const AddressCustomer = ({route, navigation}) => {
  const {address, documentId} = route.params; // params from customer screen
  const [textAddress, setTextAddress] = React.useState(address);
  const dispatch = useDispatch();
  const {user} = useSelector(states => ({user: states.auth.user}));
  const onEditAddressCustomer = async () => {
    Keyboard.dismiss();
    await dispatch(
      reduxEditCustomer({
        userId: user.uid,
        documentId: documentId,
        objectDataEdit: {
          address: textAddress,
        },
      }),
    );
    navigation.navigate('Stack', {
      screen: 'DetailCustomer',
    });
  };
  return (
    <View style={styles.container}>
      {/* view header */}
      <Header>
        <View style={styles.viewHeader}>
          <TouchableOpacity onPress={() => navigation.goBack()}>
            <IconGoBackArrow
              width={responsiveWidth(20)}
              height={responsiveHeight(20)}
            />
          </TouchableOpacity>
          <Text style={styles.txtTitleHeader}>{I18n.t('editAddress')}</Text>
          <TouchableOpacity onPress={onEditAddressCustomer}>
            <IconCheckMark
              width={responsiveWidth(20)}
              height={responsiveHeight(20)}
              fill={Colors.white}
            />
          </TouchableOpacity>
        </View>
      </Header>
      {/* view content */}
      <View style={styles.viewTextInput}>
        <Hoshi
          autoFocus={true}
          value={textAddress}
          onChangeText={text => setTextAddress(text)}
          labelStyle={styles.inputLabel}
          inputStyle={styles.inputStyle}
          label={I18n.t('address')}
          style={styles.containerInput}
          borderColor={Colors.lightGreen}
          borderHeight={2}
          inputPadding={responsiveHeight(15)}
          backgroundColor={Colors.white}
        />
      </View>
    </View>
  );
};

export default AddressCustomer;
