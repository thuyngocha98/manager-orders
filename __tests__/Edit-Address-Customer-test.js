/**
 * @format
 */

import 'react-native';
import React from 'react';
import EditAddressCustomer from '@modules/customer/EditAddressCustomer';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-orientation-locker', () => {
  return {
    getInitialOrientation: jest.fn(),
  };
});

describe('render correctly', () => {
  const props = {
    route: {
      params: {
        address: '123 Le Duan, Quan 1, Ho Chi Minh',
      },
    },
  };
  jest.useFakeTimers();
  it('render edit address customer', () => {
    const tree = renderer.create(<EditAddressCustomer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
