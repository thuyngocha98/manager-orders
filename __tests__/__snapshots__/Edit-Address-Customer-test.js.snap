// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly render edit address customer 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f0f3f8",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#116149",
        "flexDirection": "column",
        "height": 226.5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 14,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={71}
            bbWidth={18.5}
            focusable={false}
            height={71}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 71,
                  "width": 18,
                },
              ]
            }
            vbHeight={447.243}
            vbWidth={447.243}
            width={18.5}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M420.361 192.229a31.967 31.967 0 00-5.535-.41H99.305l6.88-3.2a63.998 63.998 0 0018.08-12.8l88.48-88.48c11.653-11.124 13.611-29.019 4.64-42.4-10.441-14.259-30.464-17.355-44.724-6.914a32.018 32.018 0 00-3.276 2.754l-160 160c-12.504 12.49-12.515 32.751-.025 45.255l.025.025 160 160c12.514 12.479 32.775 12.451 45.255-.063a32.084 32.084 0 002.745-3.137c8.971-13.381 7.013-31.276-4.64-42.4l-88.32-88.64a64.002 64.002 0 00-16-11.68l-9.6-4.32h314.24c16.347.607 30.689-10.812 33.76-26.88 2.829-17.445-9.019-33.88-26.464-36.71z"
                fill={4294967295}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 60.5,
              "fontWeight": "bold",
            }
          }
        >
          Sửa địa chỉ
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={71}
            bbWidth={18.5}
            fill="#FFFFFF"
            focusable={false}
            height={71}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 71,
                  "width": 18,
                },
              ]
            }
            vbHeight={78.369}
            vbWidth={78.369}
            width={18.5}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M78.049 19.015L29.458 67.606a1.094 1.094 0 01-1.548 0L.32 40.015a1.094 1.094 0 010-1.547l6.704-6.704a1.095 1.095 0 011.548 0l20.113 20.112 41.113-41.113a1.095 1.095 0 011.548 0l6.703 6.704a1.094 1.094 0 010 1.548z"
                fill={4294967295}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingTop": 18,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "borderBottomColor": "#b9c1ca",
            "borderBottomWidth": 2,
          },
          Object {
            "borderBottomColor": "#e6e8ea",
            "borderBottomWidth": 1,
          },
          Object {
            "height": 101.5,
            "width": null,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        backgroundColor="#FFFFFF"
        borderColor="#13AA52"
        borderHeight={2}
        height={48}
        inputPadding={53.5}
        inputStyle={
          Object {
            "color": "#0a1f44",
            "fontFamily": "HelveticaNeue-Medium",
            "fontSize": 53.5,
            "fontWeight": "normal",
            "height": 142.5,
            "paddingVertical": 0,
          }
        }
        label="Địa chỉ"
        labelStyle={
          Object {
            "color": "#98a1b1",
            "fontFamily": "HelveticaNeue-Medium",
            "fontSize": 53.5,
            "fontWeight": "normal",
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "bottom": 2,
              "color": "#6a7989",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 0,
              "position": "absolute",
            },
            Object {
              "color": "#0a1f44",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 53.5,
              "fontWeight": "normal",
              "height": 142.5,
              "paddingVertical": 0,
            },
            Object {
              "height": 48,
              "left": 53.5,
              "width": null,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="123 Le Duan, Quan 1, Ho Chi Minh"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 53.5,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6a7989",
                "fontSize": 16,
              },
              Object {
                "color": "#98a1b1",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 53.5,
                "fontWeight": "normal",
              },
            ]
          }
        >
          Địa chỉ
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 24,
            },
            Object {
              "backgroundColor": undefined,
              "width": 53.5,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#13AA52",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
